erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -342,6 +343,32 @@ def on_cancel(self):
 		self.unlink_sales_invoice_from_timesheets()
 		self.ignore_linked_doctypes = ('GL Entry', 'Stock Ledger Entry', 'Repost Item Valuation')
 
+	def check_sales_return_linked_with_payment_entry(self):
+		# If a Sales Return is linked with payment entry along with other invoices,
+		# the cancellation of the Sales Return causes a problem between paid and allocated amount
+
+		payment_entries = []
+		if self.is_return:
+			payment_entries = frappe.db.sql_list("""
+				SELECT
+					t1.name
+				FROM
+					`tabPayment Entry` t1, `tabPayment Entry Reference` t2
+				WHERE
+					t1.name = t2.parent
+					and t1.docstatus = 1
+					and t2.reference_name = %s
+				""", self.name)
+		if payment_entries:
+			for pe in payment_entries:
+				payment_entry = frappe.get_doc("Payment Entry", pe)
+				if frappe.db.get_single_value('Accounts Settings', 'unlink_payment_on_cancellation_of_invoice'):
+					if len(payment_entry.references) > 1:
+						frappe.throw(_("Please cancel and amend the Payment Entry {0} to unallocate the amount of \
+							this Credit Note {1}")
```

Comment:
This should be called return invoice as this msg will pop up only when cancelling return invoices linked in payment entry

Reply:
Fixed for both the scenarios